toolbox/atmospheric_tools/grib2_to_fv3_regrid.py

```diff
@@ -0,0 +1,570 @@
+                short_name = getattr(msg, 'shortName', None)
+                
+                # Check if this is an aerosol parameter
+                if self._is_aerosol_parameter(param_name, short_name):
+                    level = getattr(msg, 'level', 0)
+                    level_type = getattr(msg, 'levelType', 'unknown')
+                    
+                    # Get data and coordinates
+                    data, lats, lons = msg.data()
+                    
+                    # Store the data
+                    field_key = f"{param_name}_{level_type}_{level}"
+                    aerosol_data[field_key] = {
+                        'data': data,
+                        'lats': lats,
+                        'lons': lons,
+                        'level': level,
+                        'level_type': level_type,
+                        'param_name': param_name,
+                        'units': getattr(msg, 'units', 'unknown')
+                    }
+                    
+        except Exception as e:
+            logging.error(f"Error reading GRIB2 data: {e}")
+            raise
+            
+        return aerosol_data
+    
+    def _is_aerosol_parameter(self, param_name, short_name):
+        """
+        Check if a parameter is aerosol-related.
+        
+        Parameters:
+        -----------
+        param_name : str
+            Full parameter name
+        short_name : str
+            Short parameter name
+            
+        Returns:
+        --------
+        bool : True if parameter is aerosol-related
+        """
+        if not param_name and not short_name:
+            return False
+            
+        aerosol_keywords = [
+            'aerosol', 'dust', 'so2', 'so4', 'bc', 'oc', 'pm2.5', 'pm10',
+            'sulfate', 'nitrate', 'ammonium', 'organic', 'carbon',
+            'sea salt', 'seasalt'
+        ]
+        
+        param_lower = (param_name or '').lower()
+        short_lower = (short_name or '').lower()
+        
+        return any(keyword in param_lower or keyword in short_lower 
+                   for keyword in aerosol_keywords)
+
+
+class VerticalInterpolator:
+    """Class to handle vertical interpolation using ak/bk coordinates."""
+    
+    def __init__(self, akbk_file=None):
+        """
+        Initialize vertical interpolator.
+        
+        Parameters:
+        -----------
+        akbk_file : str, optional
+            Path to file containing ak/bk coefficients
+        """
+        self.akbk_file = akbk_file
+        self.ak = None
+        self.bk = None
+        
+    def load_akbk_coefficients(self):
+        """Load ak/bk coefficients from file."""
+        if not self.akbk_file or not os.path.exists(self.akbk_file):
+            logging.warning("No ak/bk file provided, using default coefficients")
+            # Use default L127 coefficients as example
+            self._create_default_akbk()
+            return
+            
+        try:
```

Comment:
The ak/bk values are in a netCDF file with the following format:
```
netcdf gfs_ctrl {
dimensions:
	nvcoord = 2 ;
	levsp = 129 ;
variables:
	int ntrac ;
	double vcoord(nvcoord, levsp) ;
}
```
ak is nvcoord=0 and bk is nvcoord=1